protocols/v2/codec-sv2/src/decoder.rs

```diff
@@ -23,6 +33,14 @@ use buffer_sv2::{Buffer as IsBuffer, BufferFromSystemMemory as Buffer};
 
 #[cfg(feature = "with_buffer_pool")]
 use buffer_sv2::{Buffer as IsBuffer, BufferFromSystemMemory, BufferPool};
+
+// The buffer type for holding intermediate data during decoding.
+//
+// When the `with_buffer_pool` feature is enabled, `Buffer` is a pool-allocated buffer type
```

Comment:
where is the definition of `Buffer` without the feature flag `with_buffer_pool` ?

Reply:
I added some extra language here with the last commit. Can you review now? I am a bit confused on how the code knows to default to `BufferFromSystemMemory` when the feature flag is disabled. I would expect there to be a type like:

```rs
#[cfg(not(feature = "with_buffer_pool"))]
type Buffer = Vec<u8>;
```

@fi3, do you have insights here?

Reply:
It happens in the import statement:

https://github.com/stratum-mining/stratum/blob/e65cb320173b0048ec65dc4e042d8013a7fa962d/protocols/v2/codec-sv2/src/decoder.rs#L21-L22

I have added a note in the `codec-sv2` technical debt issue #1129 to define this as a type rather than rely on the implication of the import statement.